plugins/services/src/js/containers/service-connection/SDKServiceConnectionEndpointList.js

```diff
@@ -0,0 +1,235 @@
+      this.setState({ servicePreviousState: serviceStatus });
+    }
+  }
+
+  getClipboardTrigger(command) {
+    return <EndpointClipboardTrigger command={command} />;
+  }
+
+  getJSONEndpoint(endpoint, key) {
+    return (
+      <ConfigurationMapSection key={key}>
+        <ConfigurationMapHeading>
+          {endpoint.getName()}
+        </ConfigurationMapHeading>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Address
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            <span>{this.getClipboardTrigger(endpoint.getAddress())}</span>
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            DNS
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {this.getClipboardTrigger(endpoint.getDns())}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            VIP
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {this.getClipboardTrigger(endpoint.getVip())}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+      </ConfigurationMapSection>
+    );
+  }
+
+  getFileEndpoint(endpoint, key) {
+    return (
+      <ConfigurationMapRow key={key}>
+        <ConfigurationMapLabel>
+          {endpoint.getName()}
+        </ConfigurationMapLabel>
+        <ConfigurationMapValue>
+          <a
+            className="active endpoint-download"
+            download={endpoint.getName()}
+            href={RouterUtil.getResourceDownloadPath(
+              "text/plain",
+              endpoint.getName(),
+              endpoint.getData()
+            )}
+          >
+            <span>
+              <Icon
+                id="download"
+                className="endpoint-download-icon"
+                size="mini"
+              />
+              {" "}
+              Download
+            </span>
+          </a>
+        </ConfigurationMapValue>
+      </ConfigurationMapRow>
+    );
+  }
+
+  getJSONEndpoints(endpoints) {
+    if (!Array.isArray(endpoints)) {
+      return null;
+    }
+
+    return endpoints
+      .filter(endpoint => {
+        return endpoint.isJSON();
+      })
+      .map((endpoint, index) => {
+        return this.getJSONEndpoint(endpoint, index);
+      });
+  }
+
+  getFileEndpoints(endpoints) {
+    if (!Array.isArray(endpoints)) {
+      return null;
+    }
+
+    const fileEndpoints = endpoints.filter(endpoint => {
+      return !endpoint.isJSON();
+    });
+
+    if (fileEndpoints.length === 0) {
+      return null;
+    }
+
+    return (
+      <ConfigurationMapSection>
+        <ConfigurationMapHeading>
+          Files
+        </ConfigurationMapHeading>
+        {fileEndpoints.map((endpoint, index) => {
+          return this.getFileEndpoint(endpoint, index);
```

Comment:
Not blocking as I believe that using the index in this case should be sufficient, but if possible always prefer an unique element identifier so that React can still identify the element even when you _mutate_ the array.